src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -1408,13 +1416,22 @@ private bool TryOpen(TaskCompletionSource<DbConnectionInternal> retry)
         {
             SqlConnectionString connectionOptions = (SqlConnectionString)ConnectionOptions;
 
-            // .NET Core 2.0 and up supports a Globalization Invariant Mode to reduce the size of
-            // required libraries for applications which don't need globalization support. SqlClient
-            // requires those libraries for core functionality and will throw exceptions later if they
-            // are not present. Throwing on open with a meaningful message helps identify the issue.
-            if (CultureInfo.GetCultureInfo("en-US").EnglishName.Contains("Invariant"))
+            CultureCheckState cultureCheckState = _cultureCheckState;
+            if (cultureCheckState != CultureCheckState.Standard)
             {
-                throw SQL.GlobalizationInvariantModeNotSupported();
+                // .NET Core 2.0 and up supports a Globalization Invariant Mode to reduce the size of
+                // required libraries for applications which don't need globalization support. SqlClient
+                // requires those libraries for core functionality and will throw exceptions later if they
+                // are not present. Throwing on open with a meaningful message helps identify the issue.
+                if (cultureCheckState == CultureCheckState.Unknown)
+                {
+                    cultureCheckState = CultureInfo.GetCultureInfo("en-US").EnglishName.Contains("Invariant") ? CultureCheckState.Invariant : CultureCheckState.Standard;
+                    _cultureCheckState = cultureCheckState;
```

Comment:
Why not direct assignment?

`_cultureCheckState = CultureInfo.GetCultureInfo("en-US").EnglishName.Contains("Invariant") ? CultureCheckState.Invariant : CultureCheckState.Standard;`

Reply:
Because I'm interacting with a static I split the work into phases, copy the static state to locals, work in locals to check and possibly setup new state and then if needed write the state back to the static. Written this way if atomicity is needed it's a couple of simple changes and even without atomic requirements it's clear exactly when the state can possibly be read and written to.

Reply:
But you are allocating this local variable everytime with TryOpen(), which is very important call here, isn't that something of concern for performance/memory consumption?

Reply:
Sure but it's an enum so it'll take 4 bytes in the stack frame not on the heap and doesn't need any cleanup or GC tracking, it's the same as using a bool or int.

Reply:
Yes, but if it can be avoided, we should consider that.
Memory is definitely a concern in stack.

Reply:
Changed.